cmd/nri-kubernetes/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/newrelic/infra-integrations-sdk/integration"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/sethgrid/pester"
+	"k8s.io/apimachinery/pkg/version"
+
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	"github.com/newrelic/nri-kubernetes/v2/internal/discovery"
+	"github.com/newrelic/nri-kubernetes/v2/src/apiserver"
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+	"github.com/newrelic/nri-kubernetes/v2/src/ksm"
+	"github.com/newrelic/nri-kubernetes/v2/src/metric"
+	"github.com/newrelic/nri-kubernetes/v2/src/scrape"
+	"github.com/newrelic/nri-kubernetes/v2/src/sink"
+	"github.com/newrelic/nri-kubernetes/v2/src/storage"
+)
+
+const defaultLabelSelector = "app.kubernetes.io/name=kube-state-metrics"
```

Comment:
Seems slightly awkward to me having this, which is tied to KSM, in the `main` package. Perhaps we could have an `internal/discovery/ksm` package, with a KSM-specific discoverer (which would be a very thin layer on top of the endrpoits discoverer) handling this defautls and the `getDiscoverer` funciton. WDYT?